tests/PyPI-pytest/run_all.py

```diff
@@ -0,0 +1,381 @@
+
+            if search_mode.abortIfExecuted():
+                break
+            continue
+
+
+
+        # compare outputs
+        stdout_diff = compareOutput(
+            "stdout",
+            uncompiled_stdout,
+            compiled_stdout,
+            ignore_warnings=True,
+            ignore_infos=True,
+            syntax_errors=True,
+        )
+
+        stderr_diff = compareOutput(
+            "stderr",
+            uncompiled_stderr,
+            compiled_stderr,
+            ignore_warnings=True,
+            ignore_infos=True,
+            syntax_errors=True,
+        )
+
+
+        results.append((package_name,stdout_diff,stderr_diff))
+
+        exit_code = stdout_diff or stderr_diff
+
+
+        my_print(
+            bcolors.RED if exit_code else bcolors.GREEN,
+            "\n=================================================================================",
+            bcolors.ENDC
+        )
+
+        my_print(
+            bcolors.RED if exit_code else bcolors.GREEN,
+            "--- %s ---" % package_name,
+            "exit_stdout:",
+            stdout_diff,
+            "exit_stderr:",
+            stderr_diff,
+            bcolors.ENDC
+        )
+
+        if exit_code:
+            my_print(
+                bcolors.RED,
+                "Error, outputs differed for package %s." % package_name
+            )
+        else:
+            my_print(
+                bcolors.GREEN,
+                "No differences found for package %s." % package_name
+            )
+
+        my_print(
+            "=================================================================================\n",
+            bcolors.ENDC
+        )
+
+
+        if exit_code != 0 and search_mode.abortOnFinding(dirname=None, filename=package_name):
+            break
+
+        if search_mode.abortIfExecuted():
+            break
+
+    search_mode.finish()
+
+
+    # give a summary of all packages
+
+    my_print(
+        bcolors.YELLOW,
+        "\n\n=====================================SUMMARY=====================================",
+        bcolors.ENDC
+    )
+
+    for package_name, stdout_diff, stderr_diff in results:
+        my_print(
+            bcolors.RED if (stdout_diff or stderr_diff) else bcolors.GREEN,
+            bcolors.UNDERLINE + package_name + bcolors.ENDC,
+            "-",
+
+            bcolors.RED if stdout_diff else bcolors.GREEN,
+            "stdout:",
+            stdout_diff,
+            bcolors.ENDC,
+
+            bcolors.RED if stderr_diff else bcolors.GREEN,
+            "stderr:",
+            stderr_diff,
+            bcolors.ENDC,
+
+            "\n---------------------------------------------------------------------------------"
+        )
+
+    my_print(
+        bcolors.YELLOW,
+        "TOTAL NUMBER OF PACKAGES TESTED: %s" % len(results),
+        bcolors.ENDC
+    )
+
+    num_failed = sum(y or z for _,y,z in results)
+
+    my_print(

```

Comment:
I wonder, if `my_print` should not learn about colors, with a `color = "yellow"` parameter style, what is your take on that, would make it re-usable.

Reply:
Sounds good.